### DISCUSSION TITLE
KeyDown event does not occur when applying Style.

### DISCUSSION THREAD

**Discussion by Hoeung** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3165] [Q&A] [answered]
Applying the MaterialDesignToolToggleListBox style to the ListBox does not cause a KeyDown event.
I'm trying to turn Toggle off by pressing ESC, but I can't get an event, so I'm looking for a way.
Attempted EventSetter and KeyBinding for ListBox but no event occurred.
Help me...ㅠㅠ

**Comment by nicolaihenriksen** [comment, marked as answer]
@Hoeung I suspect the reason you are not able to hit the `KeyDown` (or `PreviewKeyDown` for that matter) event handler is because the `ListBox` does not have focus; even after using the mouse to toggle an item.

I believe that is because the toggle style uses a `PreviewMouseLeftButtonDown` event to capture the mouse click (see [ListBoxAssist.cs](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/ceb3ebdbb51cb3451119b8e05c6441b13fd457f0/MaterialDesignThemes.Wpf/ListBoxAssist.cs)) and manually perform the toggle action. This also marks the event as handled which means the `ListBoxItem` probably never receives an actual `MouseDown` event which would cause the `ListBox` (or `ListBoxItem`) to be focused; this is what happens for a "normal" `ListBox`.

One solution to this, which unfortunately requires a fix in MDIX, would be to add a call to `listBoxItem.Focus()` in [this line](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/ceb3ebdbb51cb3451119b8e05c6441b13fd457f0/MaterialDesignThemes.Wpf/ListBoxAssist.cs#L37). That would ensure that once you have used the mouse to toggle an item, the `ListBox`/`ListBoxItem` will gain focus. In turn, that will allow you to capture the `KeyDown` event on the `ListBox`. @Keboo Thoughts?

**Reply by Hoeung** [reply]
Thank you so much for your answer.

For now, it was inevitable, so the problem was solved by focusing on other buttons in View with ListBox.

**Reply by Keboo** [reply]
On the question of adding the focus line. I am not opposed to adding in the focus line. I can see that code is already trying to handle various cases by forwarding the routed event down into the ripple, but that won't move the focus.
